README.md

```diff
@@ -1,7 +1,335 @@
+setSeparator(separator: string): void;
+get<T>(key: string|string[]): Promise<T>;
+getTtl(key: string|string[]): Promise<number>;
+getBuffer(key: string|string[]): Promise<Buffer>;
+set<T>(key: string|string[], value: T, ttl?: number): Promise<void>;
+setBuffer(key: string|string[], value: Buffer, ttl?: number): Promise<void>;
+del(key: string|string[]): Promise<void>;
+flush(): Promise<void>;
+keys(match: string): Promise<string[]>;
+```
+
+Refer to [ioredis](https://github.com/luin/ioredis) for more information on the functions get, getTtl, getBuffer, set, setBuffer, del, flush and keys. For getSeparator and setSeparator you simply have the ability to get and set the seperator between key parts (by default the seperator is ':'), therefore you can supply an array of key parts to all functions instead of creating the key yourself. onError is a way to set the callback when errors occurs on the redis side.
+
+## List of Common Exceptions
+
+A simple utility for abstracting exceptions.
+
+The following are included:
+
+- BadRequestException: 400
+- UnauthorizedException: 401
+- ForbiddenException: 403
+- NotFoundException: 404
+- MethodNotAllowedException: 405
+- ConflictException: 409
+- TooManyRequestsException: 429
+
+All the listed exceptions are extended from the Exception class which has the following functions and public variables available to it:
+
+```typescript
+statusCode: number;
+code: string;
+details?: any;
+body: {code: string, message: string, details?: any};
+
+fromHttpCode(httpCode: number, message: string, code?: string, details?: any): Exception;
+```
+
+## Request Validation
+
+Used to validate requests. Simply provide a validation object to the validate function as follows:
+
+```typescript
+import { RequestValidator, RequestValidation } from '@ssense/framework';
+
+const validation: RequestValidation = {
+    query: {
+        language: {
+            type: 'string',
+            required: true,
+            values: ['en', 'fr', 'ja']
+        },
+        country: {
+            type: 'string',
+            required: true
+        }
+    }
+};
+
+const validate = RequestValidator.validate(validation);
+```
+
+The function returned from RequestValidator.validate() will be another function that can be used to validate an http request. Pass in the request, response and error callback as follows:
+
+```typescript
+validate(req, res, (err?: Error) => {
+    next(err);
+});
+```
+
+For more information please check the [typescript definition file](https://github.com/SSENSE/node-framework/blob/master/index.d.ts).
+
+## Safe Shutdown Server
+
+Use this utility to wrap a standard server when creating servers in your application. It simply provides extra functionality to allow for safe shutdowns of the server. This can be useful when kubernetes makes requests to an application for graceful shutdown during autoscaling.
+
+Note that this utility can wrap both the Inversify Restify Server and the standard Express Server.
```

Comment:
You should remove Inversify here, we use it indeed in our projects to provide dependency injection, but the safe shutdown servers only wraps the restify default server.